### ISSUE TITLE
Adding album to queue ungrouped

### ISSUE THREAD

**Issue by StrangeSeasOfThought** [Taiko2k/TauonMusicBox#182]
Yeah, that again. I noticed it's already possible to do so by additionaly selecting the last track of the previous album. Almost there... ;)

![queue](https://user-images.githubusercontent.com/43582772/66233036-9fd84f80-e6ea-11e9-8e92-24d9a9b6097b.gif)

**Comment by Taiko2k** [comment]
Did you have a way of choosing this in mind? Perhaps a right click menu option in the queue named something like "Ungroup Album" would be best?

**Comment by StrangeSeasOfThought** [comment]
> Did you have a way of choosing this in mind? Perhaps a right click menu option in the queue named something like "Ungroup Album" would be best?
Well, ideally, using drag and drop (basically like above) would already ungroup the album. But I can understand if this isn't wanted...

Right click option in queue would be fine, I guess. But would it also work mid-album?

Something else I noticed: When an album is ungrouped and I click playlist menu option "add album to queue after current album", the new album lands on top of the queue. If there already are multiple tracks belonging to another album in queue, the new album should be queued after them.

**Comment by Taiko2k** [comment]
> if this isn't wanted...

Indeed I think the current behavior is prefered.

> using drag and drop (basically like above) 

Maybe a hold <kbd>Ctrl</kbd> modifier for this then?

>  the new album should be queued after them.

Okay I have made a modification. Though of note here, "current album" is the album that is playing, not necessarily what is at the top of the queue.

**Comment by StrangeSeasOfThought** [comment]
> Maybe a hold Ctrl modifier for this then?

That'd be great.

> Though of note here, "current album" is the album that is playing, not necessarily what is at the top of the queue.

Yes, of course. Thanks.

**Comment by StrangeSeasOfThought** [comment]
Something else: Sometimes I feel like an optional 'for different artist, start a new row" for gallery would be pretty neat. Not sure without seeing it in practice, though...

**Comment by Taiko2k** [comment]
> That'd be great. And Ctrl + mousewheel click for gallery?

Okay, I've pushed this change now.

> if different artist, start a new row

I'm not feeling this idea, I'm not sure it would look good, and unfortunately it would require a fair amount of rework of the gallery to allow this.

Also may be kind of undesired in situations where the artists are shuffled in the playlist. I know the artist list kind of relies on them being in order, but you can ignore it when its not needed and still looks okay, however the gallery is more a center part of navigation, so if the user were to enable this setting, its likely they'd encounter this awkward situation.

**Comment by StrangeSeasOfThought** [comment]
> > That'd be great. And Ctrl + mousewheel click for gallery?
> 
> Okay, I've pushed this change now.

Cheers!

- mwclicking the folder title line adds grouped albums to queue. Would be great for Ctrl+mwclick to work there, too.
- Please expand the add to queue (empty queue) area again. For someone who uses the queue a lot, the old way was much more convenient. Previously, I could just drag the tracks to the left and drop them. Now, I have to move to the left and down and have to actually pay attention on where to drop them, so that I don't accidentally create another playlist. :(

**Comment by Taiko2k** [comment]
> Would be great for Ctrl+mwclick to work there, too.

Okay

> should display the album title in the notification,

Okay

> Please expand the add to queue (empty queue) area again.

Okay, I've restored this setting.

**Comment by StrangeSeasOfThought** [comment]
> > Please expand the add to queue (empty queue) area again.
> 
> Okay, I've restored this setting.

The setting is back, but doesn't work.

**Comment by Taiko2k** [comment]
> The setting is back, but doesn't work.

Oh right, fixed

**Comment by StrangeSeasOfThought** [comment]
- Single track albums should always be added to queue ungrouped.

**Comment by Taiko2k** [comment]
> Single track albums should always be added to queue ungrouped.

I think for consistency, the current behavior is okay.

**Comment by StrangeSeasOfThought** [comment]
> I think for consistency, the current behavior is okay.

Hmm, it's just that grouped albums kinda imply that there's more than one track inside...

- Inside the queue: Being able to select and move multiple tracks to change their position.


(Maybe a longterm goal, e.g. for 5.x or so: Being able to expand and collapse grouped albums. Display number of tracks inside. Moving one track to another track of the same album, they join together as a group.

![queue](https://user-images.githubusercontent.com/43582772/66458091-246ff880-ea72-11e9-944d-a8ca73871016.png)

from Cantata.)

**Comment by Taiko2k** [comment]
Out of curiosity, what of these operations would you be performing?
 - Skipping tracks within an album?
 - Re-ordering tracks within an album?
 - Re-ordering tracks between albums?
 - All of the above?

**Comment by StrangeSeasOfThought** [comment]
> * Skipping tracks within an album?
> * Re-ordering tracks within an album?
> * Re-ordering tracks between albums?
> * All of the above?

On a daily basis? None of these. Only rarely, if at all.

> Also, have you considered making a playlist to function like a queue, like, drag and drop items to a playlist tab, then you can re-arrange in that playlist?

No, queues are superior because:

- Queues aren't permanent, I don't have to clean up afterwards.
- Building queues is faster: I can add tracks via a single mouse button.
- 'Split view': I can see the queue + any playlist. I can't see playlist A + playlist B.
- I can immediately decide a tracks position on queue when adding the track (via drag and drop). To do so on a playlist I'd have to hop from playlist A to B first.
- ...

**Comment by StrangeSeasOfThought** [comment]
Speaking of playlists (bug): When importing a playlist, the app creates two playlist tabs. An empty one and one with the imported tracks.

**Comment by Taiko2k** [comment]
> When importing a playlist, the app creates two playlist tabs. An empty playlist and one with the imported tracks.

Ah right, this happens when dropping it onto the left playlist panel. Ive pushed a fix now.

**Comment by StrangeSeasOfThought** [comment]
An option to not show playlists in left sidepanel when drag-and-dropping tracks to queue. In my case there are currently 12 lines of playlists, they're just in the way.

**Comment by Taiko2k** [comment]
@StrangeSeasOfThought Are you aware of the new queue only view? Or are you wanting to see the playlists normally?

I don't think changing the layout on drag is good because the user has likely already visually targeted where they want to drag the tracks to.

**Comment by StrangeSeasOfThought** [comment]
I've always had playlists disabled on the left sidepanel, even before the new queue only view. However, when using drag, the queue automatically appears but also lists all playlists again (I know this is intended, but I think it shouldn't be when queue only view is enabled).

**Comment by Taiko2k** [comment]
The intended behavior should be that the "Queue" view only and always ever shows the queue, and any setting to disable playlists from "Playlists + Queue" should be removed and no longer have effect.

You're experiencing different behaivor?

**Comment by StrangeSeasOfThought** [comment]
Yeah, I can't take a screenshot right now, but the playlists show up on the left sidepanel whenever I drag tracks to the queue. By queue I mean the one that appears when I'm e.g. in artist list view and use drag and drop.

**Comment by StrangeSeasOfThought** [comment]
I noticed an inconsistency: The queue (via MB3) has a smaller width than the artist list view and the drag and drop queue.

**Comment by Taiko2k** [comment]
@StrangeSeasOfThought The artist list is intentionally larger, but the queue shouldn't be different just from opening it a different way (also MB3 doesn't open the queue?). It does change depending of Window size though.

**Comment by StrangeSeasOfThought** [comment]
> @StrangeSeasOfThought The artist list is intentionally larger, but the queue shouldn't be different just from opening it a different way (also MB3 doesn't open the queue?). It does change depending of Window size though.

I use MB3 on the icon in the top left corner to view the queue. When the app is maximized the queue has a smaller width than the artist list view. So when I go to the queue, the left border of the playlist moves further to the left (increases in size) and when I go back to the artist list view the left border of the playlist moves to the right (decreases in size). They should've the same width imo.
